docs: linked to Explore Apps in relevant places (#94315)
grafana/grafana: docs/sources/introduction/_index.md

The following sections provide an overview of Grafana features and links to product documentation to help you learn more. For more guidance and ideas, check out our [Grafana Community forums](https://community.grafana.com/).

> ## Explore metrics, logs, traces and profiles

Reviewer: Each of the Explore apps is a product/feature name, which means they should be proper nouns. i'm not sure if we can use proper nouns in a list like this. I'll ask the other writers for input.

Reviewer: ```suggestion
## Explore metrics, logs, traces, and profiles
```

> Explore your data through ad-hoc queries and dynamic drilldown. Split view and compare different time ranges, queries and data sources side by side. Refer to [Explore]({{< relref "../explore" >}}) for more information. Or use our new suite of [Explore Apps for simplified exploration]({{< relref "../explore/simplified-exploration" >}}) to browse your telemetry data without writing queries.

Reviewer: ```suggestion
Explore your data through ad-hoc queries and dynamic drilldown.
Split view and compare different time ranges, queries and data sources side by side. 
Refer to [Explore]({{< relref "../explore" >}}) for more information. 

You can also use one of the [Explore Apps for simplified exploration]({{< relref "../explore/simplified-exploration" >}}) to browse your telemetry data without writing queries.
```
